### PR TITLE
Allow manually configuring containerd version via CLI flag

### PR DESCRIPTION
**Pull request by dajudge**
**What this PR does / why we need it**:
A [broken containerd release](https://github.com/containerd/containerd/pull/7838) made it impossible for our setup to provision working new nodes. This was a result of `machine-controller` automatically installing the latest `1.6*` release of `containerd` available.

This PR introduces the command line flag `--node-containerd-version` to manually pin the `containerd` version to install to a specific release.

The default for this new flag is empty, so the previously existing behavior remains intact if the flag is not specified.

**Which issue(s) this PR fixes**:
Implements partially https://github.com/kubermatic/kubeone/issues/2545

**What type of PR is this?**
/kind feature

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
Introduce `--node-conterainerd-version` CLI flag.
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: pkg/containerruntime/containerruntime.go
Comment (xmudrii) [inline pkg/containerruntime/containerruntime.go:140]: I don't see that we use this version anywhere. I think that we might need to add it in two places:

https://github.com/kubermatic/machine-controller/blob/e7e4ba8eb535086374108ed235c2e33bab7f4292/pkg/containerruntime/containerd.go#L67

https://github.com/kubermatic/machine-controller/blob/e7e4ba8eb535086374108ed235c2e33bab7f4292/pkg/containerruntime/docker.go#L88

**Review by xmudrii — COMMENTED** [review]
Thanks a lot for this contribution, we really appreciate it! :tada:

I have one question, but other than that, LGTM.

File: pkg/containerruntime/containerruntime.go
Comment (dajudge) [reply pkg/containerruntime/containerruntime.go:141]: https://github.com/kubermatic/machine-controller/blob/e7e4ba8eb535086374108ed235c2e33bab7f4292/pkg/containerruntime/containerd.go#L67 gets overwritten by the value passed via CLI (if present) at https://github.com/kubermatic/machine-controller/blob/e7e4ba8eb535086374108ed235c2e33bab7f4292/pkg/containerruntime/containerd.go#L71

As for the docker runtime: I added the same mechanism already in place for containerd.

WDYT?

**Comment by dajudge** [comment]
I also noticed that the containerd version gets overwritten to a legacy version in the case of AmazonLinux at the following places:

https://github.com/AtomicJar/machine-controller/blob/5d54195d47e9e78104f330bc877030b5cefd9740/pkg/containerruntime/docker.go#L98

https://github.com/AtomicJar/machine-controller/blob/5d54195d47e9e78104f330bc877030b5cefd9740/pkg/containerruntime/containerd.go#L77

Maybe it would make sense to move those `if` blocks above the `switch` to be below the latter so the value provided via CLI overrides this fallback? :thinking:

File: pkg/containerruntime/containerruntime.go
Comment (xmudrii) [reply pkg/containerruntime/containerruntime.go:141]: This looks good now!

**Comment by xmudrii** [comment]
> Maybe it would make sense to move those if blocks above the switch to be below the latter so the value provided via CLI overrides this fallback?

I don't have strong opinions. Amazon Linux is a bit specific -- they ship only one specific version of containerd (and other relevant packages). I think, for consistency reasons, it might make sense to do so.

In the meanwhile, I'll trigger tests for this PR.
/test all

**Comment by dajudge** [comment]
> Maybe it would make sense to move those if blocks above the switch to be below the latter so the value provided via CLI overrides this fallback?

To unblock this PR I've put above change on a [separate branch](https://github.com/AtomicJar/machine-controller/tree/cli-over-al2-fallback).

Once this is PR here is on `main` I'd simply open a new PR and we can discuss it in detail there.

WDYT?

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
